recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -84,15 +84,19 @@ def _build_visual(self):
         msbuild = MSBuild(self)
         msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=self._vs_configuration)
 
-    def _build_autotools_impl(self, configure_args):
+    def _build_autotools_impl(self, configure_args, host_arch=None):
         win_bash = False
         if self._is_mingw:
             win_bash = True
 
         autotools = AutoToolsBuildEnvironment(self, win_bash=win_bash)
         if self._is_mingw:
             self.run("autoreconf -i", cwd=self._source_subfolder, win_bash=win_bash)
-        autotools.configure(args=configure_args, configure_dir=self._source_subfolder, host=False)
+        host = None
+        if host_arch:
+            host = False
+            configure_args.append("--host=%s" % host_arch)
+        autotools.configure(args=configure_args, configure_dir=self._source_subfolder, host=host)
```

Comment:
> host (Optional, Defaulted to None): To specify a value for the parameter --host. If None it will try to detect the value if cross-building is detected according to the settings. If False, it will not use this argument at all.

https://docs.conan.io/en/latest/reference/build_helpers/autotools.html#configure

I am certainly no expert when it comes to cross-compiling... But if I understand the documentation the `host` would be overridden by the configure 🤔 

I would have assumed the last one _correct_ assuming libsodium uses custom host values.

Reply:
Yes, but no `host` flag was getting passed to configure in some circumstances, because the code has `host=False`, and doesn't manually set the `configure_args` to have a `--host=xxx` for all cases.

For example, in a x86_64-linux-gnu to aarch64-linux-gnu build, no host was set, because in:
 ```python
    def _build_autotools_linux(self, configure_args):
        self._build_autotools_impl(configure_args)
```
no custom logic is used. Rather than try to introduce _more_ custom logic, this PR instead allows Conan to take care of it in its standard way. Testing this on the aforementioned setup now works, where before the error in #4815 was produced.

I would think even more of the custom host setting could be removed, but this would require testing setups I don't have available to me.

Reply:
Ahhh! Thank you for the explanation, that makes much more sense now!